internal/gcsx/bucket_manager_test.go

```diff
@@ -0,0 +1,122 @@
+package gcsx
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/googlecloudplatform/gcsfuse/internal/storage"
+	"github.com/jacobsa/gcloud/gcs"
+	"github.com/jacobsa/gcloud/gcs/gcsfake"
+	. "github.com/jacobsa/ogletest"
+	"github.com/jacobsa/timeutil"
+)
+
+func TestFlags(t *testing.T) { RunTests(t) }
+
+const TestBucketName string = "gcsfuse-default-bucket"
+
+////////////////////////////////////////////////////////////////////////
+// Boilerplate
+////////////////////////////////////////////////////////////////////////
+
+type BucketManagerTest struct {
+	bucket            gcs.Bucket
+	storageHandle     *storage.Storageclient
+	fakeStorageServer storage.FakeStorage
+}
+
+var _ SetUpInterface = &BucketManagerTest{}
+var _ TearDownInterface = &BucketManagerTest{}
+
+func init() { RegisterTestSuite(&BucketManagerTest{}) }
+
+func (t *BucketManagerTest) SetUp(_ *TestInfo) {
+	var err error
+	t.storageHandle = t.fakeStorageServer.CreateStorageHandle()
+	t.bucket, err = t.storageHandle.BucketHandle(TestBucketName)
+
+	AssertEq(nil, err)
+	AssertNe(nil, t.bucket)
+}
+
+func (t *BucketManagerTest) TearDown() {
+	t.fakeStorageServer.ShutDown()
+}
+
+func (t *BucketManagerTest) TestNewBucketManagerMethod() {
+	storageClient := t.storageHandle
```

Comment:
why are you extacting it a seperate varable, you can pass t.storageHandle directly to NewBucketManager() method.

Reply:
fixed